File: backend/src/main/java/de/stecknitz/backend/web/resources/TestResource.java
package de.stecknitz.backend.web.resources;

import de.stecknitz.backend.core.service.TwelveDataService;
import de.stecknitz.backend.core.service.client.twelvedata.dto.EndOfDayDTO;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/test")
@Slf4j
@RequiredArgsConstructor
public class TestResource {

    private final TwelveDataService twelveDataService;

    @GetMapping
    public ResponseEntity<EndOfDayDTO> get() {
        log.debug("get");
        return ResponseEntity.ok(twelveDataService.get());
    }

}

File: backend/src/main/kotlin/de/stecknitz/backend/web/resources/TestResource.kt
package de.stecknitz.backend.web.resources

import de.stecknitz.backend.core.service.TwelveDataServiceKotlin
import de.stecknitz.backend.core.service.client.twelvedata.dto.EndOfDayDTOKotlin
import lombok.RequiredArgsConstructor
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/test")
@RequiredArgsConstructor
class TestResource(private val twelveDataService: TwelveDataServiceKotlin) {

    @GetMapping
    fun get(): ResponseEntity<EndOfDayDTOKotlin> {
        log.debug("get")
        return ResponseEntity.ok(twelveDataService.getEndOfDayData())
    }

    @GetMapping(path = ["/test"])
    fun getTest() {
        log.debug("getTest")
    }

    companion object {
        private val log = LoggerFactory.getLogger(TestResource::class.java)
    }
}
